xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-watch/src/main/java/org/xwiki/notifications/filters/watch/WatchedUserReference.java

```diff
@@ -41,25 +49,33 @@
  */
 public class WatchedUserReference implements WatchedEntityReference
 {
-    private String userId;
+    private final String userId;
 
-    private NotificationFilterPreferenceManager notificationFilterPreferenceManager;
+    private final ComponentManager componentManager;
 
-    private EventUserFilterPreferencesGetter preferencesGetter;
+    private final NotificationFilterPreferenceManager notificationFilterPreferenceManager;
+
+    private final EventUserFilterPreferencesGetter preferencesGetter;
+    private final UserReferenceSerializer<DocumentReference> userReferenceSerializer;
 
     /**
      * Construct a WatchedUserReference.
      * @param userId id of the user to watch.
-     * @param preferencesGetter the instance of EventUserFilterPreferencesGetter
-     * @param notificationFilterPreferenceManager the notification filter manager
-     * @since 10.9
+     * @param componentManager the component manager for loading needed components
      */
-    public WatchedUserReference(String userId, EventUserFilterPreferencesGetter preferencesGetter,
-            NotificationFilterPreferenceManager notificationFilterPreferenceManager)
+    public WatchedUserReference(String userId, ComponentManager componentManager)
     {
         this.userId = userId;
-        this.preferencesGetter = preferencesGetter;
-        this.notificationFilterPreferenceManager = notificationFilterPreferenceManager;
+        this.componentManager = componentManager;
+        try {
+            this.preferencesGetter = this.componentManager.getInstance(EventUserFilterPreferencesGetter.class);
+            this.notificationFilterPreferenceManager =
+                this.componentManager.getInstance(NotificationFilterPreferenceManager.class);
+            this.userReferenceSerializer = componentManager.getInstance(
+                new DefaultParameterizedType(null, UserReferenceSerializer.class, DocumentReference.class), "document");
+        } catch (ComponentLookupException e) {
+            throw new RuntimeException(e);
```

Comment:
Not best exception handling but not sure how to proceed...

Reply:
I'm not sure why you catch it in the first place, shouldn't WatchedUserReference(String, ComponentManager) simply throw it ?

Reply:
That's probably indeed best option, since I'm breaking the API anyway

Reply:
I would also suggest the same thing in other areas where you have the same logic, like WatchedLocationReference.